engine/engine.go

```diff
@@ -358,8 +373,26 @@ func (q *Query) Explain() *ExplainOutputNode {
 	return explainVector(q.exec)
 }
 
-func (q *Query) Profile() {
-	// TODO(bwplotka): Return profile.
+func (q *Query) Analyze() *AnalyzeOutputNode {
+	if observableRoot, ok := q.exec.(model.ObservableVectorOperator); ok {
+		return analyzeVector(observableRoot)
+	}
+	return nil
+}
+
+func analyzeVector(obsv model.ObservableVectorOperator) *AnalyzeOutputNode {
+	telemetry, obsVectors := obsv.Analyze()
+	name, _ := obsv.(model.VectorOperator).Explain()
```

Comment:
We don't need to typecast here.
```suggestion
	name, _ := obsv.Explain()
```

Reply:
Why not add the name to the original struct? This will be quite costly :sweat_smile: